### PR TITLE
[FX-5766] Fix multiline input adornment position for horizontal layout

### PR DESCRIPTION
**Pull request by ruslan-sed**
[FX-5766]

### Description

Fix multiline input adornment position for horizontal layout.

### How to test

Go to http://localhost:9001/?path=/story/forms-input--input and check the `Status horizontal` story.

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| image | image |

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- ~Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core~
- ~Annotate all `props` in component with documentation~
- [x] Create `examples` for component
- ~Ensure that deployed demo has expected results and good examples~
- ~Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).~
- [x] Self reviewed
- [x] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

**Breaking change**

- ~codemod is created and showcased in the changeset~
- ~test alpha package of Picasso in StaffPortal~

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-5766]: https://toptal-core.atlassian.net/browse/FX-5766?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ruslan-sed** [comment]
@toptal-anvil ping reviewers

File: packages/base/Input/src/Input/Input.tsx [hunk 1]
```diff
@@ -25,6 +25,7 @@ import type {
   InputIconAdornmentProps,
 } from '@toptal/picasso-input-adornment'
 import type { BaseInputProps, Status } from '@toptal/picasso-outlined-input'
+import { useFieldsLayoutContext } from '@toptal/picasso-form'
```
Comment (dmaklygin) [inline packages/base/Input/src/Input/Input.tsx:28]: 👍

File: packages/base/Input/src/Input/Input.tsx [hunk 2]
```diff
@@ -195,8 +196,11 @@ const MultilineAdornment = ({
   status,
   testIds,
 }: MultilineAdornmentProps) => {
+  const { layout } = useFieldsLayoutContext()
+  const isHorizontal = layout === 'horizontal'
```
Comment (dmaklygin) [inline packages/base/Input/src/Input/Input.tsx:200]: nit:
```suggestion
  const className = layout === 'horizontal' ?  '[grid-area:error]' : ''
```

**Comment by dmaklygin** [comment]
Temploy is here: https://picasso.toptal.net/fx-5766-multilineadornment-position-bug-fix/?path=/story/picasso-introduction--introduction

File: packages/base/Input/src/Input/story/StatusHorizontal.example.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      <Form.Field>
+        <Form.Label>Multiline Success With Counter</Form.Label>
+        <Input
+          value='Ukraine'
+          multiline
+          rows={4}
+          status='success'
+          counter='entered'
+        />
+      </Form.Field>
```
Comment (dmaklygin) [inline packages/base/Input/src/Input/story/StatusHorizontal.example.tsx:32]: I would keep only a single field to have the example lightweight

File: packages/base/Input/src/Input/Input.tsx (hunk 2 above)
Comment (ruslan-sed) [reply packages/base/Input/src/Input/Input.tsx:200]: Will remove it all

File: packages/base/Input/src/Input/story/StatusHorizontal.example.tsx (hunk 3 above)
Comment (ruslan-sed) [reply packages/base/Input/src/Input/story/StatusHorizontal.example.tsx:32]: Lets leave multiline with status and multiline with status and counter

**Review by TomasSlama — APPROVED** [review]

**Review by dmaklygin — APPROVED** [review]
Checked locally 👍

**Pull request merged**